File: DSA/02Array/Assignments/11_count_good_meals/count_good_meals.cpp
class Solution {
public:
	int countPairs(vector<int>& deliciousness) {
		// Your code goes here
		unordered_map<int,int> mp;
        long long ans = 0;
        for(auto a : deliciousness)
        {
            for(int i=1;i<=pow(2,21);i*=2)
            {
                if(mp.count(i-a))
                {
                    ans += mp[i-a];
                }
            }
            mp[a]++;
        }
        return ans % (int)(1e9 + 7);
	}
};

File: DSA/02Array/Assignments/11_count_good_meals/count_good_meals.py
#Link: https://leetcode.com/problems/count-good-meals/description/
class Solution:
	def countPairs(self, deliciousness: List[int]) -> int:
		# Your code goes here
		d={}
		ans=0
		for n in deliciousness:
			for i in range(0,22):
				ans+=d.get(2**i-n,0)
			d[n]=d.get(n,0)+1
		return ans % 1_000_000_007